src/internal/connector/exchange/service_iterators_test.go

```diff
@@ -0,0 +1,651 @@
+			scope:               allScope,
+			failFast:            true,
+			expectErr:           assert.Error,
+			expectNewColls:      0,
+			expectMetadataColls: 0,
+		},
+	}
+	for _, test := range table {
+		suite.T().Run(test.name, func(t *testing.T) {
+			ctx, flush := tester.NewContext()
+			defer flush()
+
+			collections := map[string]data.Collection{}
+
+			err := filterContainersAndFillCollections(
+				ctx,
+				qp,
+				test.getter,
+				collections,
+				statusUpdater,
+				test.resolver,
+				test.scope,
+				dps,
+				control.Options{FailFast: test.failFast},
+			)
+			test.expectErr(t, err)
+
+			// collection assertions
+
+			deleteds, news, metadatas, doNotMerges := 0, 0, 0, 0
+			for _, c := range collections {
+				if c.FullPath().Service() == path.ExchangeMetadataService {
+					metadatas++
+					continue
+				}
+
+				if c.State() == data.DeletedState {
+					deleteds++
+				}
+
+				if c.State() == data.NewState {
+					news++
+				}
+
+				if c.DoNotMergeItems() {
+					doNotMerges++
+				}
+			}
+
+			assert.Zero(t, deleteds, "deleted collections")
+			assert.Equal(t, test.expectNewColls, news, "new collections")
+			assert.Equal(t, test.expectMetadataColls, metadatas, "metadata collections")
+			assert.Equal(t, test.expectDoNotMergeColls, doNotMerges, "doNotMerge collections")
+
+			// items in collections assertions
+			for k, expect := range test.getter {
+				coll := collections[k]
+
+				if coll == nil {
+					continue
+				}
+
+				exColl, ok := coll.(*Collection)
+				require.True(t, ok, "collection is an *exchange.Collection")
+
+				assert.ElementsMatch(t, expect.added, exColl.added, "added items")
+				assert.ElementsMatch(t, expect.removed, exColl.removed, "removed items")
+			}
+		})
+	}
+}
+
+func (suite *ServiceIteratorsSuite) TestFilterContainersAndFillCollections_incrementals() {
+	var (
+		userID   = "user_id"
+		tenantID = tester.M365TenantID(suite.T())
+		cat      = path.EmailCategory // doesn't matter which one we use,
+		qp       = graph.QueryParams{
+			Category:      cat,
+			ResourceOwner: userID,
+			Credentials:   suite.creds,
+		}
+		statusUpdater = func(*support.ConnectorOperationStatus) {}
+		allScope      = selectors.NewExchangeBackup(nil).MailFolders(selectors.Any(), selectors.Any())[0]
+		commonResults = mockGetterResults{
+			added:    []string{"added"},
+			newDelta: api.DeltaUpdate{URL: "new_delta_url"},
+		}
+		expiredResults = mockGetterResults{
+			added: []string{"added"},
+			newDelta: api.DeltaUpdate{
+				URL:   "new_delta_url",
+				Reset: true,
+			},
+		}
+	)
+
+	prevPath := func(t *testing.T, at string) path.Path {
+		p, err := path.Builder{}.
+			Append(at).
+			ToDataLayerExchangePathForCategory(tenantID, userID, cat, false)
+		require.NoError(t, err)
+
+		return p
+	}
+
+	table := []struct {
+		name                  string
+		getter                mockGetter
+		resolver              graph.ContainerResolver
+		dps                   DeltaPaths
+		expectNewColls        int
```

Comment:
note: planning a small refactor that replaces a baseline count of statuses with ensuring the expected collection contains the expected status.